roles/build_push_container/tasks/cleanup.yml

```diff
@@ -0,0 +1,33 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Stop and remove registry container
+  containers.podman.podman_container:
+    name: registry
+    state: absent
+
+- name: List images created and used by role
+  ansible.builtin.command:
+    cmd: >-
+      podman images -q
+      --filter=label=build_push_role=true
+      --filter=reference=docker.io/library/registry:2
+  register: podman_images_output
+
+- name: Remove images created and used by role
+  ansible.builtin.command:
+    cmd: podman rmi -f {{ item }}
+  loop: "{{ podman_images_output.stdout_lines }}"
```

Comment:
Try to a look at the [podman_prune](https://docs.ansible.com/ansible/latest/collections/containers/podman/podman_prune_module.html) module (written by yours truly :D)

You should be able to replace those tasks with a single task :)

```
- name: Cleanup podman resources created by this role
  containers.podman.podman_prune:
    container: true
    container_filters:
      name: registry
    image: true
    image_filters:
      label: build_push_role=true
      reference: docker.io/library/registry:2
```

I just wrote it without testing, but you should have an idea!